### PR TITLE
Add dev app to test managed identity on app service

### PR DESCRIPTION
**Pull request by neha-bhargava**
Fixes #
Add a dev app for managed identity app service

**Changes proposed in this request**

**Testing**

**Performance impact**

**Documentation**
- [ ] All relevant documentation is updated.

### CODE REVIEW

File: ManagedIdentityWebApi/ManagedIdentityWebApi.csproj [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Identity.Client" Version="4.51.0" />
```
Comment (gladjohn) [inline ManagedIdentityWebApi/ManagedIdentityWebApi.csproj:10]: we need a identity client project reference so we can continue to test our code. No direct reference to NuGet

File: ManagedIdentityWebApi/Controllers/AppServiceController.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            _logger.LogInformation("Get token from MSAL for managed identity.");
+            try
+            {
+                IManagedIdentityApplication mi = CreateManagedIdentityApplication(userAssignedId);
+
+                var result = await mi.AcquireTokenForManagedIdentity(resourceUri).ExecuteAsync().ConfigureAwait(false);
+
+                return "Access token received. Token Source: " + result.AuthenticationResultMetadata.TokenSource;
+            }
+            catch (MsalException ex)
```
Comment (gladjohn) [inline ManagedIdentityWebApi/Controllers/AppServiceController.cs:46]:
```suggestion
            catch (MsalException ex)
```
thought we had a managedidentityexception? no?

**Review by gladjohn — APPROVED** [review]
just one comment on direct project reference

File: ManagedIdentityWebApi/Controllers/AppServiceController.cs (hunk 2 above)
Comment (neha-bhargava) [reply ManagedIdentityWebApi/Controllers/AppServiceController.cs:46]: No we need to add one

Comment (gladjohn) [reply ManagedIdentityWebApi/Controllers/AppServiceController.cs:46]: Ok if we do not have a tracking item for the items we agreed with Azure SDK, we should create those

File: ManagedIdentityWebApi/ManagedIdentityWebApi.csproj [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    <ImplicitUsings>enable</ImplicitUsings>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Swashbuckle.AspNetCore" Version="6.2.3" />
+    <PackageReference Include="Swashbuckle.AspNetCore.Annotations" Version="6.5.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\src\client\Microsoft.Identity.Client\Microsoft.Identity.Client.csproj" />
```
Comment (gladjohn) [inline ManagedIdentityWebApi/ManagedIdentityWebApi.csproj:15]: looks good

**Comment by gladjohn** [comment]
Please merge when build succeeds so @trwalke can use this app for his hack project

**Review by bgavrilMS — APPROVED** [review]

**Pull request merged**